### PR TITLE
chore(sql): added test with asymmetric pools

### PR DESCRIPTION
**Pull request by bluestreak01**
I wanted to verify that if we use multiple pools we don't share object instances between them

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: core/src/test/java/io/questdb/griffin/engine/table/AsyncFilteredRecordCursorFactoryTest.java [hunk 1]
```diff
@@ -668,50 +762,57 @@ public boolean getCloneSymbolTables() {
...
+                    "s\tt\n",
+                    factory,
+                    true,
+                    true,
+                    false,
+                    false,
+                    sqlExecutionContext
+            );
+
+            // !!! This call is the problematic one !!!
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/griffin/engine/table/AsyncFilteredRecordCursorFactoryTest.java:792]: This comment looks like a leftover from one of previous PRs.

**Review by puzpuzpuz — APPROVED** [review]

File: core/src/test/java/io/questdb/griffin/engine/table/AsyncFilteredRecordCursorFactoryTest.java (hunk 1 above)
Comment (bluestreak01) [reply core/src/test/java/io/questdb/griffin/engine/table/AsyncFilteredRecordCursorFactoryTest.java]: fixed in https://github.com/questdb/questdb/pull/2140/commits/8864a2d55b7aa9268003847388c0277b3d6fab9a

**Comment by bluestreak01** [comment]
@puzpuzpuz I also refactored tests some more to remove even more duplicate code around WorkerPools

**Review by puzpuzpuz — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by ideoma** [comment]
#### [PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)
